commit/types.go

```diff
@@ -58,173 +49,14 @@ func DecodeCommitPluginObservation(encodedObservation []byte) (Observation, erro
-
-	// A map from chain selectors to each chain's consensus f (failure tolerance)
-	FChain map[cciptypes.ChainSelector]int
-}
-
-// GasPricesArray returns a list of gas prices
-func (co ConsensusObservation) GasPricesArray() []cciptypes.GasPriceChain {
-	gasPrices := make([]cciptypes.GasPriceChain, 0, len(co.GasPrices))
-	for chain, gasPrice := range co.GasPrices {
-		gasPrices = append(gasPrices, cciptypes.NewGasPriceChain(gasPrice.Int, chain))
-	}
-	sort.Slice(gasPrices, func(i, j int) bool { return gasPrices[i].ChainSel < gasPrices[j].ChainSel })
-
-	return gasPrices
-}
-
-// TokenPricesArray returns a list of token prices
-func (co ConsensusObservation) TokenPricesArray() []cciptypes.TokenPrice {
-	tokenPrices := make([]cciptypes.TokenPrice, 0, len(co.TokenPrices))
-	for tokenID, tokenPrice := range co.TokenPrices {
-		tokenPrices = append(tokenPrices, cciptypes.NewTokenPrice(tokenID, tokenPrice.Int))
-	}
-	sort.Slice(tokenPrices, func(i, j int) bool { return tokenPrices[i].TokenID < tokenPrices[j].TokenID })
-
-	return tokenPrices
-}
-
-type OutcomeType int
-
-const (
-	ReportIntervalsSelected OutcomeType = iota + 1
-	ReportGenerated
-	ReportEmpty
-	ReportInFlight
-	ReportTransmitted
-	ReportTransmissionFailed
-)
-
 type Outcome struct {
-	OutcomeType                     OutcomeType                 `json:"outcomeType"`
-	RangesSelectedForReport         []plugintypes.ChainRange    `json:"rangesSelectedForReport"`
-	RootsToReport                   []cciptypes.MerkleRootChain `json:"rootsToReport"`
-	OffRampNextSeqNums              []plugintypes.SeqNumChain   `json:"offRampNextSeqNums"`
-	TokenPrices                     []cciptypes.TokenPrice      `json:"tokenPrices"`
-	GasPrices                       []cciptypes.GasPriceChain   `json:"gasPrices"`
-	ReportTransmissionCheckAttempts uint                        `json:"reportTransmissionCheckAttempts"`
-}
-
-// Sort all fields of the given Outcome
-func (o Outcome) sort() {
-	sort.Slice(o.RangesSelectedForReport, func(i, j int) bool {
-		return o.RangesSelectedForReport[i].ChainSel < o.RangesSelectedForReport[j].ChainSel
-	})
-	sort.Slice(o.RootsToReport, func(i, j int) bool {
-		return o.RootsToReport[i].ChainSel < o.RootsToReport[j].ChainSel
-	})
-	sort.Slice(o.OffRampNextSeqNums, func(i, j int) bool {
-		return o.OffRampNextSeqNums[i].ChainSel < o.OffRampNextSeqNums[j].ChainSel
-	})
-	sort.Slice(o.TokenPrices, func(i, j int) bool {
-		return o.TokenPrices[i].TokenID < o.TokenPrices[j].TokenID
-	})
-	sort.Slice(o.GasPrices, func(i, j int) bool {
-		return o.GasPrices[i].ChainSel < o.GasPrices[j].ChainSel
-	})
+	MerkleOutcome merkleroot.Outcome `json:"merkleOutcome"`
+	TokensOutcome tokenprice.Outcome `json:"tokensOutcome"`
+	FeesOutcome   chainfee.Outcome   `json:"gasOutcome"`
 }
 
 // Encode encodes an Outcome deterministically
 func (o Outcome) Encode() ([]byte, error) {
-
-	// Sort all lists to ensure deterministic serialization
-	o.sort()
-
```

Comment:
Feels error prone to remove this, it will ensure that the outcome is always deterministic